Makefile

```diff
@@ -168,7 +168,7 @@ generate-documentation:
 
 lint:
 	# This version number must be kept in sync with CI workflow lint one.
-	docker run --rm -v $(shell pwd):/app -w /app golangci/golangci-lint:v1.46.2 golangci-lint run --fix
+	docker run --rm --user $(shell id -u):$(shell id -g) -v $(shell pwd):/app -w /app golangci/golangci-lint:v1.46.2 golangci-lint run --fix
```

Comment:
This change is needed indeed because the fixed files also change their owner to `root`. However, in my case, after this change it is not working:

```bash
$ make lint
# This version number must be kept in sync with CI workflow lint one.
docker run --rm --user 1000:1000 -v /home/jose/Documents/kinvolk/src/inspektor-gadget:/app -w /app golangci/golangci-lint:v1.46.2 golangci-lint run --fix
2022/07/05 15:05:26 failed to initialize build cache at /.cache/golangci-lint: mkdir /.cache: permission denied
make: *** [Makefile:171: lint] Error 1
```

Is there something wrong in my env?